### PR TITLE
integ-cli: Skip some unix-specific cli tests

### PR DESCRIPTION
**Pull request by ahmetb**
Skipping some of the tests closely tied to running in a
unix environment. Windows does not support chmod/chown
and this causes some tests to fail creating desired
behavior.
- `TestBuildWithInaccessibleFilesInContext`: uses chown/chmod
- `TestBuildDockerfileOutsideContext`: uses os.Symlink, not implemented on
  windows
- `TestCpUnprivilegedUser`: uses chmod, and requires 'unprivilegeduser'
  created by Dockerfile (and thus requires to run inside container)
- `TestBuildChownSingleFile`: uses chown

I wonder if I can write windows equivalents of those using golang
pkg/os calls for windows. Most of them are not implemented, especially
windows-specific security ACLs etc. 

Signed-off-by: Ahmet Alp Balkan ahmetalpbalkan@gmail.com
cc: @unclejack @tianon @jfrazelle @icecrime @tiborvass

### CODE REVIEW

File: integration-cli/test_vars_windows.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// +build windows
+
+package main
+
+const (
+	// idetifies if test suite is running on a unix platform
+	isUnixCli = false
```
Comment (LK4D4) [inline integration-cli/test_vars_windows.go]: Hmm, seems like it can be just `isUnix`?

File: integration-cli/test_vars_windows.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+// +build windows
```
Comment (LK4D4) [inline integration-cli/test_vars_windows.go]: I think you don't need tag if you use suffix.

File: integration-cli/test_vars_unix.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// +build !windows
```
Comment (LK4D4) [inline integration-cli/test_vars_unix.go]: I see that in standard library tags:
```
// +build darwin dragonfly freebsd linux nacl netbsd openbsd solaris
```
treated as `unix`, but maybe `!windows` is the same :) apart from android tag.

Comment (ahmetalpbalkan) [inline integration-cli/test_vars_unix.go]: Just following the conversation from rest of the code base. This is probably fine and if it's not it's will be a compilation error which can be easily fixed.

File: integration-cli/test_vars_windows.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline integration-cli/test_vars_windows.go]: While porting docker cli to windows we decided to explicitly keep those tags explicitly even though file names do the same thing.

File: integration-cli/test_vars_windows.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline integration-cli/test_vars_windows.go]: Being more explicit never hurts, does it? :-)

Comment (ahmetalpbalkan) [inline integration-cli/test_vars_windows.go]: Also at some point I can see us running against windows daemon, so isUnix would probably get refactored to this anyway to clarify what it actually refers to.

Comment (LK4D4) [inline integration-cli/test_vars_windows.go]: Was pretty implicit for me actually :P But now I got idea.

**Comment by LK4D4** [comment]
LGTM

**Comment by ahmetalpbalkan** [comment]
@LK4D4 you made me notice a typo. Can you please re-LGTM? :)

File: integration-cli/requirements.go [hunk 4]
```diff
@@ -17,6 +17,10 @@ var (
 		func() bool { return isLocalDaemon },
 		"Test requires docker daemon to runs on the same machine as CLI",
 	}
+	UnixCli = TestRequirement{
+		func() bool { return isLocalDaemon },
```
Comment (LK4D4) [inline integration-cli/requirements.go]: hmmmm `isLocalDaemon`?

Comment (ahmetalpbalkan) [inline integration-cli/requirements.go]: damn

**Comment by LK4D4** [comment]
LGTM

**Comment by jfrazelle** [comment]
lgtm

**Pull request merged**